### PR TITLE
Bringing back end into the front

### PR DESCRIPTION
**Pull request by treyholt**
tldr: 
Ultimately if you can pull this branch. Add at least one top from Zara.com to your cart. Go to cart and receive suggestions for the first item in your cart, it's a success. 

**NEW:::Cloud Functions**
Not seen in this PR is the added 'server-less' layer that allows us to connect our Vision API Code to our client application. There are some very particular security rules around how you are allowed to engage with cloud vision, but by keeping all interactions with the API in the cloud functions, we have universal access to our service accounts and can manage authentication far simpler. 

Essentially the cloud function brings the ProductSet.py functionality into a place it can be triggered and used from an https request. 
The cloud function takes a url to an image, and returns 5 result items that are most similar to that item. 

**New in Service.js**

Service.js now facilitates triggering the cloud function, parsing the results, and using it to access our firebase database to pull all the info we need into our application. 

_Authentication_
We've talked about auth a bit in the past, but since we don't have any sign in UI, and don't really have a reason for it, I figured it might still be nice to authenticate anonymously for each session. This way, we can actually by default still learn about a user's habits and potentially track things for them, but don't need them to do any login. 

**New in App.js**

What I figured would be a trivial task (extracting the image of the item to be searched from the cart and sending it to the service layer) ended up being an incredibly strange adventure. 
Zara evidently uses some form of lazy loading whereby the images or the 'src' attributes of 'img' objects aren't present on document load. This means that at runtime for our chrome extension, it's impossible to grab the link to the image in the cart. And since we can't wait on any object in particular, it seemed like we were simply shit out of luck. 
But after a lot of detective work I realized that all Zara images are served with the exact same host, and other objects around the vicinity of the image had key fields that were used to construct the image url. Once I realized that all the info was there, it was just a tedious matter of extracting the right portion from the right places. Unfortunately one piece just didn't add up. It was different on each image I looked at, different some times just between execution of the program, and not to be found anywhere in any html. Finally just out of random trial and error I realized that the missing piece specified the width of the image. And without it, no image could load. Adding that in, I was able to Frankenstein the image url and pass it along to the cloud function.

### CODE REVIEW

File: Suzie/app.js [hunk 1]
```diff
@@ -6,14 +6,27 @@ var extractCartItemsFromZara = function() {
...
+            // Welcome to html clunk hell: 
+            let hosting_path = "https://static.zara.net/photos//";
+            let image_directory = shop_cart_item_images[i].attributes[2].value;
+            let width_specification = "/w/600/";
+            let image_id = shop_cart_item_images[i].attributes[3].value;
+            let file_type = ".jpg?ts=";
+            let time_stamp = shop_cart_item_images[i].attributes[4].value;
+
+            let full_image_url = hosting_path + image_directory + width_specification + image_id + file_type + time_stamp;
+            console.log("Full image Url is: " + full_image_url);
```
Comment (treyholt) [inline Suzie/app.js:27]: dont print

**Review by Matias-A — APPROVED** [review]
